plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -1883,6 +1925,473 @@ public void testQueryTableFunction()
+                                                        .build())
+                                        .put("string_field", "sample")
+                                        .put("int_field", 2)
+                                        .put("object_field_2", ImmutableMap.<String, Object>builder()
+                                                        .put("array_string_field", ImmutableList.<String>builder()
+                                                                .addAll(Stream.of("trino", "the", "lean", "machine-ohs")::iterator)
+                                                                .build())
+                                                        .put("string_field2", "sample")
+                                                        .put("int_field", 33)
+                                                        .put("object_field_3", ImmutableMap.<String, Object>builder()
+                                                                        .put("array_string_field", ImmutableList.<String>builder()
+                                                                                .addAll(Stream.of("trino", "the", "lean", "machine-ohs")::iterator)
+                                                                                .build())
+                                                                        .put("string_field3", "some value")
+                                                                        .put("int_field3", 55)
+                                                                .buildOrThrow())
+                                                .buildOrThrow())
+                                .buildOrThrow())
+                        .buildOrThrow());
+
+        HashMap<String, Object> innerRecord = new HashMap<>();
+        innerRecord.put("object_field_inner", null);
+        index(tableName, ImmutableMap.<String, Object>builder()
+                .put("long_field", 11122233L)
+                .put("object_field_outer", innerRecord)
+                .buildOrThrow());
+
+        assertThat(query("select id_field, object_field.object_field_2.object_field_3.string_field3 from " + tableName + " where  object_field_outer.int_field_outer=44"))
+                .skippingTypesCheck()
+                .matches("VALUES ('564e6982-88ee-4498-aa98-df9e3f6b6109', 'some value')");
+        assertThat(query("select object_field_outer.int_field_outer from " + tableName + " where  object_field_outer.int_field_outer=44"))
+                .matches("VALUES CAST(44 as BIGINT)");
+        assertThat(query("select long_field, id_field, object_field.object_field_2.object_field_3.string_field3, object_field_outer.object_field_inner.int_field_inner from " + tableName + " where  long_field=11122233"))
+                .skippingTypesCheck()
+                .matches("VALUES (CAST(11122233 AS BIGINT), NULL, NULL, NULL)");
+        deleteIndex(tableName);
+    }
+
```

Comment:
The indentation of added tests look wrong.

Reply:
That's how it looks when auto spaced with.